### PR TITLE
qdate: fix tests failing in different timezones

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Tests would fail when running in different timezones due to
QDate being converted to and from QDateTime.

Change-Id: I214c4d21125d33ae0a106447dc4c7d41ce5b6142

### CODE REVIEW

**Comment by ahayzen-kdab** [comment]
(this also nicely shows the duplication of tests we have, which #81 removes :-) )

**Comment by Be-ing** [comment]
I think I encountered this bug running tests locally on macOS although it works on GitHub Actions. Are there any issues holding this PR back?

**Comment by ahayzen-kdab** [comment]
> I think I encountered this bug running tests locally on macOS although it works on GitHub Actions. Are there any issues holding this PR back?

It's waiting for review from colleagues who are on holiday :-) Hopefully will land in the next few weeks.

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
Please wait to merge this, the tests still fails on my machine!

**Review by LeonMatthesKDAB — APPROVED** [review]
Tests now pass.

**Comment by Be-ing** [comment]
The test is failing for me locally on Linux:

```
41/66 Test: example_qml_features_qttypes_test
Command: "/home/be/sw/cxx-qt/build/examples/qml_features/qttypes/example_qml_features_qttypes_test"
Directory: /home/be/sw/cxx-qt/build/examples/qml_features/qttypes
"example_qml_features_qttypes_test" start time: Apr 08 09:49 CDT
Output:
----------------------------------------------------------
********* Start testing of qttypes *********
Config: Using QtTest library 5.15.2, Qt 5.15.2 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 11.2.1 20210728 (Red Hat 11.2.1-1)), fedora 35
PASS   : qttypes::MockQtTypesTests::initTestCase()
PASS   : qttypes::MockQtTypesTests::test_color_invokable()
PASS   : qttypes::MockQtTypesTests::test_color_invokable_property()
PASS   : qttypes::MockQtTypesTests::test_color_property()
FAIL!  : qttypes::MockQtTypesTests::test_qdate_invokable() Compared values are not the same
   Actual   (): 30
   Expected (): 31
   Loc: [/home/be/sw/cxx-qt/build/examples/qml_features/qttypes/tst_qttypes.qml(109)]
FAIL!  : qttypes::MockQtTypesTests::test_qdate_invokable_property() Compared values are not the same
   Actual   (): 2021
   Expected (): 2022
   Loc: [/home/be/sw/cxx-qt/build/examples/qml_features/qttypes/tst_qttypes.qml(121)]
FAIL!  : qttypes::MockQtTypesTests::test_qdate_property() Compared values are not the same
   Actual   (): 2021
   Expected (): 2022
   Loc: [/home/be/sw/cxx-qt/build/examples/qml_features/qttypes/tst_qttypes.qml(90)]
```

**Comment by ahayzen-kdab** [comment]
> 

Thanks, I've reported #123 to track it.

**Pull request merged**
